common/app/views/fragments/atoms/structureddata/recipe.scala.html

```diff
@@ -56,20 +62,43 @@
     }
 }
 
+
 @*
   TODO - ingredient unit could be improved 1/8 instead of 0.125 for instance depending on the unit 
 *@
 @ingredientValues(ingredients: Seq[com.gu.contentatom.thrift.atom.recipe.Ingredient]) = @{
     ingredients.map { ingredient => 
         val q = if (ingredient.quantity.isDefined) {
-            ingredient.quantity.get    
+            formatQuantity(ingredient.quantity.get)    
         } else if (ingredient.quantityRange.isDefined) {
-            s"${ingredient.quantityRange.get.from}-${ingredient.quantityRange.get.to}"
+            val range = ingredient.quantityRange.get
+            s"${formatQuantity(range.from)}-${formatQuantity(range.to)}"
         } else {
             ""
         }
-        s""" ${q} ${ingredient.unit.getOrElse("")} ${ingredient.item}"""
+        s"""${q} ${formatUnit(ingredient.unit.getOrElse(""))} ${ingredient.item}"""
     }
 }
 
+@formatUnit(unit: String) = @{
+        unit match {
+            case "dsp" => "dessert spoon"
+            case "tsp" => "teaspoon"
+            case "tbsp" => "tablespoon" 
+            case _ => unit
+        }
+}
 
+@formatQuantity(q: Double) = @{
+    if(q == q.toInt) {
+        q.toInt
+    } else {
+        q match {
+            case 0.75 => "3/4"
+            case 0.5 => "1/2"
+            case 0.25 => "1/4"
+            case 0.125 => "1/8"
+            case _ => q 
+        }
```

Comment:
What about an additional match case in order to handle the whole number case? rather than an `if`
```
...
case 0.125 => "1/8"
case qty if qty == qty.toInt => qty.toInt 
case _ => q 
...
```